File: backend/backend/Models/Entities/Deposits/Deposit.cs
using backend.Models.Entities.Customers;
using backend.Models.Entities.Rooms;

namespace backend.Models.Entities.Deposits;

public class Deposit : AuditedEntity<Guid>
{
    public Guid RoomId { get; set; }
    public string CustomerName { get; set; }
    public string PhoneNumber { get; set; }
    public decimal DepositAmount { get; set; }
    public DateTime DepositDate { get; set; }
    public string Note { get; set; }
    public Guid? CustomerId { get; set; }

    /// <summary>
    /// Ngày dự kiến nhận phòng
    /// </summary>
    public DateTime? ExpectedDate { get; set; }

    public int MaximumDays { get; set; }
    public string Status { get; set; }
    public Room Room { get; set; }
    public Customer Customer { get; set; }
}

File: frontend/src/page/deposits/type/deposit.ts
import { Identifier } from '~/types/shared';

export interface DepositDto {
    id: Identifier;
    roomId: string;
    customerName: string;
    phoneNumber: string;
    depositAmount: number;
    expectedDate: Date;
    depositDate: Date;
    maximumDays: number;
    status: string;
    note: string;
}
export interface DepositGridDto extends DepositDto {
    houseName: string;
    roomCode: string;
}
